### PR TITLE
Propagate datanode version through node service

### PR DESCRIPTION
**Pull request by todvora**
Make exact datanode version available through node service pings. The version should be then used to verify that datanodes and graylog servers run the same compatible version.

Requested in https://github.com/Graylog2/graylog2-server/issues/18783
Fixes https://github.com/Graylog2/graylog2-server/issues/19680

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/cluster/nodes/DataNodeDto.java [hunk 1]
```diff
@@ -63,6 +65,11 @@ public abstract class DataNodeDto extends NodeDto {
     @JsonProperty(FIELD_CERT_VALID_UNTIL)
     public abstract Date getCertValidUntil();
 
+    @jakarta.annotation.Nullable
```
Comment (moesterheld) [inline graylog2-server/src/main/java/org/graylog2/cluster/nodes/DataNodeDto.java:68]: I think the other `@Nullable`s should also be changed here

**Review by moesterheld — CHANGES_REQUESTED** [review]

File: graylog2-server/src/main/java/org/graylog2/cluster/nodes/DataNodeDto.java (hunk 1 above)
Comment (moesterheld) [reply graylog2-server/src/main/java/org/graylog2/cluster/nodes/DataNodeDto.java:68]: from javax to jakarta, i mean

Comment (todvora) [reply graylog2-server/src/main/java/org/graylog2/cluster/nodes/DataNodeDto.java:67]: Thanks, fixed!

**Comment by todvora** [comment]
@gally47 Thanks, good point, added:

image

**Review by moesterheld — CHANGES_REQUESTED** [review]
found one more minor point, otherwise everything looks good:

for completeness, I would also propagate the version already in the `registerServer` method of the `NodePingPeriodical`

**Review by gally47 — APPROVED** [review]
Frontend LGTM. Thanks for working on this @todvora.

**Review by moesterheld — APPROVED** [review]
lgtm. thank you

**Pull request merged**